entity-framework/core/modeling/custom-function-mapping.md

```diff
@@ -0,0 +1,202 @@
+---
+title: Custom function mapping - EF Core
+description: Mapping user-defined functions to database functions
+author: maumar
+ms.date: 11/23/2020
+uid: core/modeling/custom-function-mapping
+---
+# Custom function mapping
+
+EF Core allows for using user-defined SQL functions in queries. To do that, the functions need to be mapped to a CLR method during model configuration. When translating the LINQ query to SQL, user-defined function will be called instead of the CLR function it has been mapped to.
+
+## Mapping method to a SQL function
+
+To illustrate the custom function mapping, lets define the following entities:
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/Blog.cs#Entity)]
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/Post.cs#Entity)]
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/Tag.cs#Entity)]
+
+And the following model configuration:
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/BloggingContext.cs#EntityConfiguration)]
+
+Blog can have many posts, each post can be tagged with multiple tags, and each tag can be associated with multiple posts: many-to-many relationship.
+
+Next, create the user-defined function `DistinctTagsCountForBlogPosts`, which returns the count of unique tags associated with all the posts of a given blog, based on the blog `Id`:
+
+```sql
+CREATE FUNCTION dbo.DistinctTagsCountForBlogPosts(@id int)
+    RETURNS int
+    AS
+    BEGIN
+        RETURN (SELECT COUNT(*) FROM(
+            SELECT DISTINCT t.TagId FROM dbo.Tags AS t
+            JOIN dbo.PostTag AS pt ON t.TagId = pt.TagId
+            JOIN dbo.Posts AS p ON p.PostId = pt.PostId
+            JOIN dbo.Blogs AS b ON b.BlogId = p.BlogId
+            WHERE b.BlogId = @id) AS subquery);
+    END
+```
+
+To use it in EF Core, define the following CLR method, which we will map to the user-defined function:
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/BloggingContext.cs#BasicFunctionDefinition)]
+
+In the example, the method is defined on `DbContext`, but it can also be defined in other places.
```

Comment:
It has to be static if defined elsewhere. Only on DbContext we use instance method.